### ISSUE TITLE
Getting error (__init__() got an unexpected keyword argument 'user_agent') in devops pipeline while using below command

### ISSUE THREAD

**Issue by joshimahesh** [Azure/azure-cli#18791]
## Describe the bug

**Command Name**
`az extension add`

**Errors:**
```
  File "C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\site-packages\azure\cli\core\_profile.py", line 939, in find_from_service_principal_id
    result = self._find_using_specific_tenant(tenant, token_entry[_ACCESS_TOKEN])
  File "C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\site-packages\azure\cli\core\_profile.py", line 1033, in _find_using_specific_tenant
    client = self._arm_client_factory(token_credential)
  File "C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\site-packages\azure\cli\core\_profile.py", line 881, in create_arm_client_factory
    base_url=self.cli_ctx.cloud.endpoints.resource_manager, **client_kwargs)
TypeError: __init__() got an unexpected keyword argument 'user_agent'
To open an issue, please run: 'az feedback'
##[error]Error Code: [1]
##[debug]Processed: ##vso[task.issue type=error;]Error Code: [1]
##[error]Error: Azure login failed
##[debug]Processed: ##vso[task.issue type=error;]Error: Azure login failed
##[debug]task result: Failed
##[error]Script failed with error: ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: __init__() got an unexpected keyword argument 'user_agent'
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\site-packages\knack\cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\site-packages\azure\cli\core\commands\__init__.py", line 657, in execute
    raise ex

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az extension add -n {}`

## Expected Behavior
as i am running CLI command through pipeline and logged in already this error should not appear
## Environment Summary
```
Linux-5.4.0-1051-azure-x86_64-with-debian-10.2 (Cloud Shell)
Python 3.6.10
Installer: DEB

azure-cli 2.25.0 *
azure-cli-ml 1.29.0

Extensions:
azure-cli-ml 1.29.0
ai-examples 0.2.5
ssh 0.1.5

```

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by jiasli** [comment]
Seems to be the same issue as #18626, related to Machine Learning pipelines.

**Comment by joshimahesh** [comment]
@jiasli Thanks for your reply, issue is resolved by adding 'azure-mgmt-resource==18.0.0' in my requirements.txt.

**Comment by jiasli** [comment]
As stated in https://github.com/Azure/azure-cli/issues/18626#issuecomment-879609185, adding `azure-mgmt-resource==18.0.0` will break `azureml-core` which requires `azure-mgmt-resource>=1.2.1,<15.0.0`.
